spec/factories/users.rb

```diff
@@ -4,5 +4,15 @@
     family_name  { "Doe" }
     email { "test@test.test" }
     password { "validpassword" }
+    permissions { UserRolePermissions.new("", nil) }
+  end
+
+  factory :component_user, class: User do
+    given_name { "Jane" }
+    family_name  { "Smith" }
+    email { "test@test.test" }
+    password { "validpassword" }
+    roles { "certmgr" }
```

Comment:
not sure about the naming, we're giving the user the `certmgr` role but we call the factory `:component_user`?